### PR TITLE
Partially fix #188.  Add missing api documentation to readme.

### LINKED ISSUE #188: Can't manage to have a string get analysed by brain.js
A GIF or MEME to give some spice of the internet

## *What* is wrong?
The training network doesn't train, it returns NaN for the data that it should have analysed.

## *Where* does it happen?
In the app.js run through node, after receiving data from the Twitter API

## *How* do we replicate the issue?

1. Use the dummy data below
```javascript
const trainingData =[  
   {  
      input:'RT @ObamaFoundation: This week—50 years since Dr. Martin Luther King, Jr. was killed—@BarackObama and @RepJohnLewis sat down with a group o…',
      output:{  
         Barack:1
      }
   },
   {  
      input:'Incredible to have a Chicago team in the Final Four. I’ll take that over an intact bracket any day! Congratulations to everybody @LoyolaChicago - let’s keep it going!',
      output:{  
         Barack:1
      }
   },
   {  
      input:'In Singapore with young people who are advocating for education, empowering young women, and getting involved all over Southeast Asia with a profoundly optimistic commitment to building the world they want to see. ',
      output:{  
         Barack:1
      }
   },
   {  
      input:'Very thankful for President Xi of China’s kind words on tarrifs and automobile barriers...also, his enlightenment on intellectual property and technology transfers. We will make great progress together!',
      output:{  
         Donald:1
      }
   },
   {  
      input:'Last night, it was my great honor to host America’s senior defense and military leaders for dinner at the White House. America’s military is the GREATEST fighting force in the history of the world. They all have my pledge of unwavering commitment to our men and women in uniform! ',
      output:{  
         Donald:1
      }
   },
   {  
      input:'A TOTAL WITCH HUNT!!!',
      output:{  
         Donald:1
      }
   }
]
```
2. Convert the input using encode
```javascript
function encode(arg) {
return arg.split('').map(x => (x.charCodeAt(0) / 400));
}
```

3. Run it through net.trainAsync()

## *How* important is this (1-5)?
A 4, since I think this should be able to work, but it could be my bad too!

## Expected behavior (i.e. solution)
The data should have been analysed, but somewhere inbetween something went wrong!

## Other Comments
I'm think the encoding method might not be correct here.

### PR DESCRIPTION
**Pull request by robertleeplummerjr**
A GIF or MEME to give some spice of the internet

## Description
This adds some missing documentation.

## Motivation and Context
It works well, is unit tested, and yet nobody knows about it... Till now!

## How Has This Been Tested?
Yes, but way prior to its release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +125,39 @@ net.train([{input: { r: 0.03, g: 0.7 }, output: { black: 1 }},
...
+* Be an array of values
+* Be a string
+* Have an `input` and an `output`
+  * Either of which can an array of values or a string
+
+CAUTION: When using an array of values, you can use ANY value, however, the values are represented in the neural network by a single input.  So the more _distinct values_ has _the larger your input layer_.  If you have a hundreds, thousands, or millions of floating point values _THIS IS NOT THE RIGHT CLASS FOR THE JOB_.  Also, when deviating from strings, this gets into beta
+
+Example using direct strings:
+```javascript
+var net = new brain.NeuralNetwork();
```
Comment (freddyC) [inline README.md]: Is this the correct way to initialize an `RNN`, `LSTM` or a `GRU`?  It seems like `new brain.NeuralNetwork();` is a copy paste mistake.

File: README.md [hunk 2]
```diff
@@ -123,6 +125,39 @@ net.train([{input: { r: 0.03, g: 0.7 }, output: { black: 1 }},
...
+  'ray, a drop of golden sun',
+  'me, a name I call myself',
+]);
+
+var output = net.run('doe');  // ', a deer, a female deer'
+```
+
+Example using strings with inputs and outputs:
+```javascript
+var net = new brain.NeuralNetwork();
```
Comment (freddyC) [inline README.md]: Same as above comment.

**Review by freddyC — CHANGES_REQUESTED** [review]
Let me know if I am wrong on that assumption I haven't used these network types that often.

**Comment by robertleeplummerjr** [comment]
Updated.  Ty for catching!

**Comment by robertleeplummerjr** [comment]
Since changes were addressed, going to go ahead and merge so we can handle a few outstanding pr's.

**Pull request merged**